File: HienMau/app/src/main/java/com/example/hienmau/models/YeuCauMau.java
package com.example.hienmau.models;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.util.List;

public class YeuCauMau implements Serializable {
    @SerializedName("ID")
    public int id;
    @SerializedName("NguoiDangID")
    public int nguoiDangId;

    @SerializedName("BenhVienID")
    public Integer benhVienId;

    @SerializedName("TenBenhVien")
    public String tenBenhVien;

    @SerializedName("DiaChiBenhVien")
    public String diaChiBenhVien;

    @SerializedName("ViDo")
    public Double viDo;

    @SerializedName("KinhDo")
    public Double kinhDo;
    @SerializedName("SoNguoiCan")
    public Integer soNguoiCan;
    @SerializedName("NoiDung")
    public String noiDung;

    @SerializedName("TrangThai")
    public int trangThai;

    @SerializedName("NgayDang")
    public String ngayDang;
    @SerializedName("LoaiTin")
    public int loaiTin;
    @SerializedName("ChiTiets")
    public List<YeuCauMauChiTiet> chiTiets;

    @SerializedName("NguoiDang")
    public NguoiDung nguoiDang;

    @SerializedName("NgayBatDau")
    public String ngayBatDau;

    @SerializedName("NgayKetThuc")
    public String ngayKetThuc;

    @SerializedName("GioBatDau")
    public String gioBatDau;

    @SerializedName("GioKetThuc")
    public String gioKetThuc;

    @SerializedName("BenhVien")
    public CoSoYTe benhVien;

    @SerializedName("SoNguoiDaXacNhan")
    public int soNguoiDaXacNhan;
}

File: QuanLyHienMauApi/QuanLyHienMauApi/Models/YeuCauMau.cs
using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace QuanLyHienMauApi.Models
{
    public class YeuCauMau
    {
        [Key]
        public int ID { get; set; }

        public int NguoiDangID { get; set; }

        // Thông tin bệnh viện linh hoạt
        public int? BenhVienID { get; set; }
        public string? TenBenhVien { get; set; }
        public string? DiaChiBenhVien { get; set; }
        public double? ViDo { get; set; }
        public double? KinhDo { get; set; }

        public string? NoiDung { get; set; }
        public int? SoNguoiCan { get; set; }

        [NotMapped] 
        public int SoNguoiDaXacNhan { get; set; }
        public int TrangThai { get; set; } = 0;
        public DateTime NgayDang { get; set; } = DateTime.Now;
        public string? NgayBatDau { get; set; }
        public string? NgayKetThuc { get; set; }
        public string? GioBatDau { get; set; }
        public string? GioKetThuc { get; set; }
        public int LoaiTin { get; set; } = 0;


        [ForeignKey("NguoiDangID")]
        [ValidateNever]
        public virtual NguoiDung? NguoiDang { get; set; }

        [ForeignKey("BenhVienID")]
        [ValidateNever]
        public virtual CoSoYTe? BenhVien { get; set; }

        public virtual ICollection<YeuCauMauChiTiet> ChiTiets { get; set; } = new List<YeuCauMauChiTiet>();
    }
}
